src/test/py/bazel/bzlmod/bzlmod_query_test.py

```diff
@@ -166,7 +166,7 @@ def testGenQueryTargetLiteralInGenRule(self):
     self.assertIsNotNone(output_file)
     output = output_file.readlines()
     output_file.close()
-    self.assertListEqual(['@my_repo//:lib_aaa\n'], output)
+    self.assertListEqual(['@aaa~1.0//:lib_aaa\n'], output)
```

Comment:
I am not sure how we should handle genrule:
* `legacy` mode results in these labels that would never resolve when passed in on the command-line.
* `canonical` mode would be a breaking change for WORKSPACE builds as it would add a leading `@`.
* `apparent` mode doesn't do that, but aren't as tractable for automated tooling due to the use of apparent names.